{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Running Nextflow from Colab\n",
        "\n",
        "This is a guide with code to be able to run nf-core pipelines from colab notebooks."
      ],
      "metadata": {
        "id": "J_AfYl8IOo-q"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "9g5_k5j2OnCg",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "846a491d-d618-4cd6-a0bb-20ed6d3c088f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[33m\r0% [Working]\u001b[0m\r            \rGet:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n",
            "Get:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n",
            "Hit:3 https://cli.github.com/packages stable InRelease\n",
            "Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Get:5 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [2,123 kB]\n",
            "Hit:6 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:7 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Hit:9 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Get:11 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,824 kB]\n",
            "Hit:12 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:13 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [5,969 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,594 kB]\n",
            "Get:16 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [9,431 kB]\n",
            "Get:17 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,289 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [3,856 kB]\n",
            "Get:19 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [3,526 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [6,168 kB]\n",
            "Fetched 37.2 MB in 7s (5,210 kB/s)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "44 packages can be upgraded. Run 'apt list --upgradable' to see them.\n",
            "\u001b[1;33mW: \u001b[0mSkipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\u001b[0m\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  ca-certificates-java fonts-dejavu-core fonts-dejavu-extra java-common\n",
            "  libatk-wrapper-java libatk-wrapper-java-jni libpcsclite1 libxt-dev libxtst6\n",
            "  libxxf86dga1 openjdk-17-jdk-headless openjdk-17-jre openjdk-17-jre-headless\n",
            "  x11-utils\n",
            "Suggested packages:\n",
            "  default-jre pcscd libxt-doc openjdk-17-demo openjdk-17-source visualvm\n",
            "  libnss-mdns fonts-ipafont-gothic fonts-ipafont-mincho fonts-wqy-microhei\n",
            "  | fonts-wqy-zenhei fonts-indic mesa-utils\n",
            "The following NEW packages will be installed:\n",
            "  ca-certificates-java fonts-dejavu-core fonts-dejavu-extra java-common\n",
            "  libatk-wrapper-java libatk-wrapper-java-jni libpcsclite1 libxt-dev libxtst6\n",
            "  libxxf86dga1 openjdk-17-jdk openjdk-17-jdk-headless openjdk-17-jre\n",
            "  openjdk-17-jre-headless x11-utils\n",
            "0 upgraded, 15 newly installed, 0 to remove and 44 not upgraded.\n",
            "Need to get 125 MB of archives.\n",
            "After this operation, 287 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 java-common all 0.72build2 [6,782 B]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libpcsclite1 amd64 1.9.5-3ubuntu1 [19.8 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jre-headless amd64 17.0.16+8~us1-0ubuntu1~22.04.1 [48.3 MB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 ca-certificates-java all 20190909ubuntu1.2 [12.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxt-dev amd64 1:1.2.1-1 [396 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jre amd64 17.0.16+8~us1-0ubuntu1~22.04.1 [232 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jdk-headless amd64 17.0.16+8~us1-0ubuntu1~22.04.1 [71.3 MB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jdk amd64 17.0.16+8~us1-0ubuntu1~22.04.1 [1,522 kB]\n",
            "Fetched 125 MB in 10s (12.7 MB/s)\n",
            "Selecting previously unselected package java-common.\n",
            "(Reading database ... 125082 files and directories currently installed.)\n",
            "Preparing to unpack .../00-java-common_0.72build2_all.deb ...\n",
            "Unpacking java-common (0.72build2) ...\n",
            "Selecting previously unselected package libpcsclite1:amd64.\n",
            "Preparing to unpack .../01-libpcsclite1_1.9.5-3ubuntu1_amd64.deb ...\n",
            "Unpacking libpcsclite1:amd64 (1.9.5-3ubuntu1) ...\n",
            "Selecting previously unselected package openjdk-17-jre-headless:amd64.\n",
            "Preparing to unpack .../02-openjdk-17-jre-headless_17.0.16+8~us1-0ubuntu1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-17-jre-headless:amd64 (17.0.16+8~us1-0ubuntu1~22.04.1) ...\n",
            "Selecting previously unselected package ca-certificates-java.\n",
            "Preparing to unpack .../03-ca-certificates-java_20190909ubuntu1.2_all.deb ...\n",
            "Unpacking ca-certificates-java (20190909ubuntu1.2) ...\n",
            "Selecting previously unselected package fonts-dejavu-core.\n",
            "Preparing to unpack .../04-fonts-dejavu-core_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-core (2.37-2build1) ...\n",
            "Selecting previously unselected package fonts-dejavu-extra.\n",
            "Preparing to unpack .../05-fonts-dejavu-extra_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-extra (2.37-2build1) ...\n",
            "Selecting previously unselected package libxtst6:amd64.\n",
            "Preparing to unpack .../06-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n",
            "Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Selecting previously unselected package libxxf86dga1:amd64.\n",
            "Preparing to unpack .../07-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n",
            "Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Selecting previously unselected package x11-utils.\n",
            "Preparing to unpack .../08-x11-utils_7.7+5build2_amd64.deb ...\n",
            "Unpacking x11-utils (7.7+5build2) ...\n",
            "Selecting previously unselected package libatk-wrapper-java.\n",
            "Preparing to unpack .../09-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n",
            "Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n",
            "Preparing to unpack .../10-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n",
            "Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libxt-dev:amd64.\n",
            "Preparing to unpack .../11-libxt-dev_1%3a1.2.1-1_amd64.deb ...\n",
            "Unpacking libxt-dev:amd64 (1:1.2.1-1) ...\n",
            "Selecting previously unselected package openjdk-17-jre:amd64.\n",
            "Preparing to unpack .../12-openjdk-17-jre_17.0.16+8~us1-0ubuntu1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-17-jre:amd64 (17.0.16+8~us1-0ubuntu1~22.04.1) ...\n",
            "Selecting previously unselected package openjdk-17-jdk-headless:amd64.\n",
            "Preparing to unpack .../13-openjdk-17-jdk-headless_17.0.16+8~us1-0ubuntu1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-17-jdk-headless:amd64 (17.0.16+8~us1-0ubuntu1~22.04.1) ...\n",
            "Selecting previously unselected package openjdk-17-jdk:amd64.\n",
            "Preparing to unpack .../14-openjdk-17-jdk_17.0.16+8~us1-0ubuntu1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-17-jdk:amd64 (17.0.16+8~us1-0ubuntu1~22.04.1) ...\n",
            "Setting up java-common (0.72build2) ...\n",
            "Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Setting up libxt-dev:amd64 (1:1.2.1-1) ...\n",
            "Setting up fonts-dejavu-core (2.37-2build1) ...\n",
            "Setting up libpcsclite1:amd64 (1.9.5-3ubuntu1) ...\n",
            "Setting up fonts-dejavu-extra (2.37-2build1) ...\n",
            "Setting up x11-utils (7.7+5build2) ...\n",
            "Setting up libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up ca-certificates-java (20190909ubuntu1.2) ...\n",
            "head: cannot open '/etc/ssl/certs/java/cacerts' for reading: No such file or directory\n",
            "Adding debian:Amazon_Root_CA_2.pem\n",
            "Adding debian:TeliaSonera_Root_CA_v1.pem\n",
            "Adding debian:ePKI_Root_Certification_Authority.pem\n",
            "Adding debian:AffirmTrust_Premium.pem\n",
            "Adding debian:DigiCert_High_Assurance_EV_Root_CA.pem\n",
            "Adding debian:Actalis_Authentication_Root_CA.pem\n",
            "Adding debian:QuoVadis_Root_CA_2_G3.pem\n",
            "Adding debian:UCA_Extended_Validation_Root.pem\n",
            "Adding debian:GTS_Root_R4.pem\n",
            "Adding debian:SSL.com_EV_Root_Certification_Authority_RSA_R2.pem\n",
            "Adding debian:Entrust_Root_Certification_Authority.pem\n",
            "Adding debian:GlobalSign_ECC_Root_CA_-_R4.pem\n",
            "Adding debian:Hellenic_Academic_and_Research_Institutions_ECC_RootCA_2015.pem\n",
            "Adding debian:Go_Daddy_Class_2_CA.pem\n",
            "Adding debian:Hellenic_Academic_and_Research_Institutions_RootCA_2015.pem\n",
            "Adding debian:Microsoft_RSA_Root_Certificate_Authority_2017.pem\n",
            "Adding debian:Autoridad_de_Certificacion_Firmaprofesional_CIF_A62634068.pem\n",
            "Adding debian:USERTrust_ECC_Certification_Authority.pem\n",
            "Adding debian:XRamp_Global_CA_Root.pem\n",
            "Adding debian:DigiCert_Assured_ID_Root_CA.pem\n",
            "Adding debian:TunTrust_Root_CA.pem\n",
            "Adding debian:ISRG_Root_X2.pem\n",
            "Adding debian:Entrust_Root_Certification_Authority_-_G4.pem\n",
            "Adding debian:QuoVadis_Root_CA_2.pem\n",
            "Adding debian:DigiCert_Global_Root_CA.pem\n",
            "Adding debian:Starfield_Root_Certificate_Authority_-_G2.pem\n",
            "Adding debian:Entrust_Root_Certification_Authority_-_EC1.pem\n",
            "Adding debian:emSign_Root_CA_-_C1.pem\n",
            "Adding debian:Comodo_AAA_Services_root.pem\n",
            "Adding debian:TUBITAK_Kamu_SM_SSL_Kok_Sertifikasi_-_Surum_1.pem\n",
            "Adding debian:GlobalSign_Root_E46.pem\n",
            "Adding debian:QuoVadis_Root_CA_1_G3.pem\n",
            "Adding debian:SecureTrust_CA.pem\n",
            "Adding debian:Telia_Root_CA_v2.pem\n",
            "Adding debian:D-TRUST_EV_Root_CA_1_2020.pem\n",
            "Adding debian:D-TRUST_Root_Class_3_CA_2_2009.pem\n",
            "Adding debian:Security_Communication_ECC_RootCA1.pem\n",
            "Adding debian:ISRG_Root_X1.pem\n",
            "Adding debian:Izenpe.com.pem\n",
            "Adding debian:Trustwave_Global_Certification_Authority.pem\n",
            "Adding debian:Certum_Trusted_Network_CA_2.pem\n",
            "Adding debian:Amazon_Root_CA_4.pem\n",
            "Adding debian:AffirmTrust_Commercial.pem\n",
            "Adding debian:vTrus_Root_CA.pem\n",
            "Adding debian:T-TeleSec_GlobalRoot_Class_2.pem\n",
            "Adding debian:T-TeleSec_GlobalRoot_Class_3.pem\n",
            "Adding debian:CFCA_EV_ROOT.pem\n",
            "Adding debian:COMODO_Certification_Authority.pem\n",
            "Adding debian:DigiCert_Assured_ID_Root_G3.pem\n",
            "Adding debian:COMODO_ECC_Certification_Authority.pem\n",
            "Adding debian:Buypass_Class_3_Root_CA.pem\n",
            "Adding debian:Go_Daddy_Root_Certificate_Authority_-_G2.pem\n",
            "Adding debian:SecureSign_RootCA11.pem\n",
            "Adding debian:Security_Communication_Root_CA.pem\n",
            "Adding debian:Certum_Trusted_Root_CA.pem\n",
            "Adding debian:GTS_Root_R2.pem\n",
            "Adding debian:IdenTrust_Public_Sector_Root_CA_1.pem\n",
            "Adding debian:Secure_Global_CA.pem\n",
            "Adding debian:SwissSign_Gold_CA_-_G2.pem\n",
            "Adding debian:SSL.com_EV_Root_Certification_Authority_ECC.pem\n",
            "Adding debian:AC_RAIZ_FNMT-RCM.pem\n",
            "Adding debian:Hongkong_Post_Root_CA_3.pem\n",
            "Adding debian:HiPKI_Root_CA_-_G1.pem\n",
            "Adding debian:SSL.com_Root_Certification_Authority_RSA.pem\n",
            "Adding debian:Certum_Trusted_Network_CA.pem\n",
            "Adding debian:emSign_ECC_Root_CA_-_C3.pem\n",
            "Adding debian:NAVER_Global_Root_Certification_Authority.pem\n",
            "Adding debian:Amazon_Root_CA_1.pem\n",
            "Adding debian:TWCA_Root_Certification_Authority.pem\n",
            "Adding debian:QuoVadis_Root_CA_3_G3.pem\n",
            "Adding debian:e-Szigno_Root_CA_2017.pem\n",
            "Adding debian:Atos_TrustedRoot_2011.pem\n",
            "Adding debian:OISTE_WISeKey_Global_Root_GC_CA.pem\n",
            "Adding debian:Starfield_Services_Root_Certificate_Authority_-_G2.pem\n",
            "Adding debian:Certainly_Root_E1.pem\n",
            "Adding debian:NetLock_Arany_=Class_Gold=_FÅ‘tanÃºsÃ­tvÃ¡ny.pem\n",
            "Adding debian:CA_Disig_Root_R2.pem\n",
            "Adding debian:GLOBALTRUST_2020.pem\n",
            "Adding debian:Buypass_Class_2_Root_CA.pem\n",
            "Adding debian:OISTE_WISeKey_Global_Root_GB_CA.pem\n",
            "Adding debian:Trustwave_Global_ECC_P256_Certification_Authority.pem\n",
            "Adding debian:ACCVRAIZ1.pem\n",
            "Adding debian:GDCA_TrustAUTH_R5_ROOT.pem\n",
            "Adding debian:DigiCert_TLS_ECC_P384_Root_G5.pem\n",
            "Adding debian:certSIGN_Root_CA_G2.pem\n",
            "Adding debian:SwissSign_Silver_CA_-_G2.pem\n",
            "Adding debian:COMODO_RSA_Certification_Authority.pem\n",
            "Adding debian:DigiCert_Assured_ID_Root_G2.pem\n",
            "Adding debian:GlobalSign_Root_CA.pem\n",
            "Adding debian:Entrust.net_Premium_2048_Secure_Server_CA.pem\n",
            "Adding debian:TWCA_Global_Root_CA.pem\n",
            "Adding debian:Microsec_e-Szigno_Root_CA_2009.pem\n",
            "Adding debian:USERTrust_RSA_Certification_Authority.pem\n",
            "Adding debian:Certum_EC-384_CA.pem\n",
            "Adding debian:GTS_Root_R3.pem\n",
            "Adding debian:Certigna_Root_CA.pem\n",
            "Adding debian:emSign_Root_CA_-_G1.pem\n",
            "Adding debian:D-TRUST_BR_Root_CA_1_2020.pem\n",
            "Adding debian:IdenTrust_Commercial_Root_CA_1.pem\n",
            "Adding debian:SSL.com_Root_Certification_Authority_ECC.pem\n",
            "Adding debian:DigiCert_TLS_RSA4096_Root_G5.pem\n",
            "Adding debian:AC_RAIZ_FNMT-RCM_SERVIDORES_SEGUROS.pem\n",
            "Adding debian:vTrus_ECC_Root_CA.pem\n",
            "Adding debian:HARICA_TLS_RSA_Root_CA_2021.pem\n",
            "Adding debian:SZAFIR_ROOT_CA2.pem\n",
            "Adding debian:UCA_Global_G2_Root.pem\n",
            "Adding debian:AffirmTrust_Premium_ECC.pem\n",
            "Adding debian:DigiCert_Global_Root_G3.pem\n",
            "Adding debian:ANF_Secure_Server_Root_CA.pem\n",
            "Adding debian:D-TRUST_Root_Class_3_CA_2_EV_2009.pem\n",
            "Adding debian:Trustwave_Global_ECC_P384_Certification_Authority.pem\n",
            "Adding debian:GlobalSign_Root_CA_-_R6.pem\n",
            "Adding debian:Amazon_Root_CA_3.pem\n",
            "Adding debian:DigiCert_Global_Root_G2.pem\n",
            "Adding debian:Security_Communication_RootCA2.pem\n",
            "Adding debian:Baltimore_CyberTrust_Root.pem\n",
            "Adding debian:Certainly_Root_R1.pem\n",
            "Adding debian:GlobalSign_Root_CA_-_R3.pem\n",
            "Adding debian:Security_Communication_RootCA3.pem\n",
            "Adding debian:GTS_Root_R1.pem\n",
            "Adding debian:certSIGN_ROOT_CA.pem\n",
            "Adding debian:HARICA_TLS_ECC_Root_CA_2021.pem\n",
            "Adding debian:Microsoft_ECC_Root_Certificate_Authority_2017.pem\n",
            "Adding debian:GlobalSign_ECC_Root_CA_-_R5.pem\n",
            "Adding debian:Entrust_Root_Certification_Authority_-_G2.pem\n",
            "Adding debian:DigiCert_Trusted_Root_G4.pem\n",
            "Adding debian:QuoVadis_Root_CA_3.pem\n",
            "Adding debian:GlobalSign_Root_R46.pem\n",
            "Adding debian:Starfield_Class_2_CA.pem\n",
            "Adding debian:AffirmTrust_Networking.pem\n",
            "Adding debian:emSign_ECC_Root_CA_-_G3.pem\n",
            "Adding debian:Certigna.pem\n",
            "Adding debian:CommScope_Public_Trust_RSA_Root-02.pem\n",
            "Adding debian:CommScope_Public_Trust_ECC_Root-02.pem\n",
            "Adding debian:BJCA_Global_Root_CA1.pem\n",
            "Adding debian:BJCA_Global_Root_CA2.pem\n",
            "Adding debian:Sectigo_Public_Server_Authentication_Root_R46.pem\n",
            "Adding debian:TrustAsia_Global_Root_CA_G3.pem\n",
            "Adding debian:Atos_TrustedRoot_Root_CA_ECC_TLS_2021.pem\n",
            "Adding debian:Atos_TrustedRoot_Root_CA_RSA_TLS_2021.pem\n",
            "Adding debian:CommScope_Public_Trust_ECC_Root-01.pem\n",
            "Adding debian:Sectigo_Public_Server_Authentication_Root_E46.pem\n",
            "Adding debian:SSL.com_TLS_RSA_Root_CA_2022.pem\n",
            "Adding debian:CommScope_Public_Trust_RSA_Root-01.pem\n",
            "Adding debian:TrustAsia_Global_Root_CA_G4.pem\n",
            "Adding debian:SSL.com_TLS_ECC_Root_CA_2022.pem\n",
            "done.\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.1 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for ca-certificates (20240203~22.04.1) ...\n",
            "Updating certificates in /etc/ssl/certs...\n",
            "0 added, 0 removed; done.\n",
            "Running hooks in /etc/ca-certificates/update.d...\n",
            "\n",
            "done.\n",
            "done.\n",
            "Setting up openjdk-17-jre-headless:amd64 (17.0.16+8~us1-0ubuntu1~22.04.1) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/java to provide /usr/bin/java (java) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jpackage to provide /usr/bin/jpackage (jpackage) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/keytool to provide /usr/bin/keytool (keytool) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/rmiregistry to provide /usr/bin/rmiregistry (rmiregistry) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/lib/jexec to provide /usr/bin/jexec (jexec) in auto mode\n",
            "Processing triggers for mailcap (3.70+nmu1ubuntu1) ...\n",
            "Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Setting up openjdk-17-jre:amd64 (17.0.16+8~us1-0ubuntu1~22.04.1) ...\n",
            "Setting up openjdk-17-jdk-headless:amd64 (17.0.16+8~us1-0ubuntu1~22.04.1) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jar to provide /usr/bin/jar (jar) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jarsigner to provide /usr/bin/jarsigner (jarsigner) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javac to provide /usr/bin/javac (javac) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javadoc to provide /usr/bin/javadoc (javadoc) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javap to provide /usr/bin/javap (javap) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jcmd to provide /usr/bin/jcmd (jcmd) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdb to provide /usr/bin/jdb (jdb) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdeprscan to provide /usr/bin/jdeprscan (jdeprscan) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdeps to provide /usr/bin/jdeps (jdeps) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jfr to provide /usr/bin/jfr (jfr) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jimage to provide /usr/bin/jimage (jimage) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jinfo to provide /usr/bin/jinfo (jinfo) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jlink to provide /usr/bin/jlink (jlink) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jmap to provide /usr/bin/jmap (jmap) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jmod to provide /usr/bin/jmod (jmod) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jps to provide /usr/bin/jps (jps) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jrunscript to provide /usr/bin/jrunscript (jrunscript) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jshell to provide /usr/bin/jshell (jshell) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstack to provide /usr/bin/jstack (jstack) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstat to provide /usr/bin/jstat (jstat) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstatd to provide /usr/bin/jstatd (jstatd) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/serialver to provide /usr/bin/serialver (serialver) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jhsdb to provide /usr/bin/jhsdb (jhsdb) in auto mode\n",
            "Setting up openjdk-17-jdk:amd64 (17.0.16+8~us1-0ubuntu1~22.04.1) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jconsole to provide /usr/bin/jconsole (jconsole) in auto mode\n"
          ]
        }
      ],
      "source": [
        "!apt update\n",
        "!apt install openjdk-17-jdk\n",
        "!export JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64\n",
        "!export PATH=$JAVA_HOME/bin:$PATH\n",
        "!source ~/.bashrc"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Installing Nextflow"
      ],
      "metadata": {
        "id": "ovA2dn4BPIaY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget -qO- https://get.nextflow.io | bash # Download Nextflow\n",
        "!mv nextflow /usr/bin/nextflow # Move to a path Colab can access\n",
        "!chmod +x /usr/bin/nextflow # Make it executable\n",
        "!nextflow -v # Test it"
      ],
      "metadata": {
        "id": "y3Zl5FQNPMPB",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "5739688c-0de0-4a71-c3b9-bf594fb2fd39"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[K\n",
            "      N E X T F L O W\n",
            "      version 25.10.0 build 10289\n",
            "      created 22-10-2025 16:26 UTC \n",
            "      cite doi:10.1038/nbt.3820\n",
            "      http://nextflow.io\n",
            "\n",
            "\n",
            "Nextflow installation completed. Please note:\n",
            "- the executable file `nextflow` has been created in the folder: /content\n",
            "- you may complete the installation by moving it to a directory in your $PATH\n",
            "\n",
            "nextflow version 25.10.0.10289\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Setting up Conda"
      ],
      "metadata": {
        "id": "t3wv9ZG-PRbU"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -q condacolab # -q here means quite\n",
        "import condacolab\n",
        "condacolab.install()"
      ],
      "metadata": {
        "id": "yYeM8d99PT-z",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "6de655aa-7f7b-4835-b930-7c9aca467b84"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "â¬ Downloading https://github.com/jaimergp/miniforge/releases/download/24.11.2-1_colab/Miniforge3-colab-24.11.2-1_colab-Linux-x86_64.sh...\n",
            "ðŸ“¦ Installing...\n",
            "ðŸ“Œ Adjusting configuration...\n",
            "ðŸ©¹ Patching environment...\n",
            "â² Done in 0:00:11\n",
            "ðŸ” Restarting kernel...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda config --add channels bioconda\n",
        "!conda config --add channels conda-forge\n",
        "!conda config --add channels defaults\n",
        "!conda config --set channel_priority strict"
      ],
      "metadata": {
        "id": "kk9FMckIPUjp",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ed8fe118-d206-457e-efcf-362eb04dc667"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Warning: 'bioconda' already in 'channels' list, moving to the top\n",
            "Warning: 'conda-forge' already in 'channels' list, moving to the top\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Running a Test pipeline"
      ],
      "metadata": {
        "id": "sursu7ahPZSw"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#! nextflow pull nf-core/demo"
      ],
      "metadata": {
        "id": "lMymtYBoO7Eb"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#! nextflow run nf-core/demo -profile conda,test --outdir demo-results"
      ],
      "metadata": {
        "id": "IygCHi6ePtQv"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Running QuantMS pipeline\n",
        "\n",
        "![QuantMS pipeline](https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/proteomics/notebooks/img/quantms.png)"
      ],
      "metadata": {
        "id": "z_2UONcTik6t"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Creating Data Folders for Metagenomics Analysis"
      ],
      "metadata": {
        "id": "oK3h5HyYWa2G"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir proteomics\n",
        "!mkdir proteomics/data\n",
        "!mkdir proteomics/databases"
      ],
      "metadata": {
        "id": "CZRNWpZ_T2Ke",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "05ae3748-9749-4ec0-899d-388ebce86b9c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "mkdir: cannot create directory â€˜proteomicsâ€™: File exists\n",
            "mkdir: cannot create directory â€˜proteomics/dataâ€™: File exists\n",
            "mkdir: cannot create directory â€˜proteomics/databasesâ€™: File exists\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Downloading Sample Sheet and Database File for nf-core/QuantMS"
      ],
      "metadata": {
        "id": "tabULdAGWhtD"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! wget https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/proteomics/data/experiment_design.tsv -O proteomics/data/sample_sheet.tsv\n",
        "! wget https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/proteomics/database/proteindatabase.fasta -O proteomics/databases/proteindatabase.fasta"
      ],
      "metadata": {
        "id": "Gyh474X9UtE2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ee0cbc7b-cd6f-44ee-87b9-22572e110de7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-11-07 12:32:41--  https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/proteomics/data/experiment_design.tsv\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.108.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 308 [text/plain]\n",
            "Saving to: â€˜proteomics/data/sample_sheet.tsvâ€™\n",
            "\n",
            "\r          proteomic   0%[                    ]       0  --.-KB/s               \rproteomics/data/sam 100%[===================>]     308  --.-KB/s    in 0s      \n",
            "\n",
            "2025-11-07 12:32:41 (6.54 MB/s) - â€˜proteomics/data/sample_sheet.tsvâ€™ saved [308/308]\n",
            "\n",
            "--2025-11-07 12:32:41--  https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/proteomics/database/proteindatabase.fasta\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.110.133, 185.199.111.133, 185.199.109.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.110.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 13968216 (13M) [text/plain]\n",
            "Saving to: â€˜proteomics/databases/proteindatabase.fastaâ€™\n",
            "\n",
            "proteomics/database 100%[===================>]  13.32M  37.6MB/s    in 0.4s    \n",
            "\n",
            "2025-11-07 12:32:43 (37.6 MB/s) - â€˜proteomics/databases/proteindatabase.fastaâ€™ saved [13968216/13968216]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! wget https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/proteomics/low_resources.config -O proteomics/low_resources.config"
      ],
      "metadata": {
        "id": "OXl5sLPeVVef",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "20a7122a-614a-4988-a7d5-b77e9824f61a"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-11-07 12:52:31--  https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/proteomics/low_resources.config\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.108.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 8425 (8.2K) [text/plain]\n",
            "Saving to: â€˜proteomics/low_resources.configâ€™\n",
            "\n",
            "proteomics/low_reso 100%[===================>]   8.23K  --.-KB/s    in 0.001s  \n",
            "\n",
            "2025-11-07 12:52:31 (8.17 MB/s) - â€˜proteomics/low_resources.configâ€™ saved [8425/8425]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://zenodo.org/records/17549994/files/160513-SM-AHYMJ-14.mzML?download=1 -O proteomics/data/160513-SM-AHYMJ-14.mzML\n",
        "!wget https://zenodo.org/records/17549994/files/160914_SM-9SD9G_284.mzML?download=1 -O proteomics/data/160914_SM-9SD9G_284.mzML\n",
        "!wget https://zenodo.org/records/17549994/files/160920_SM-AKTWT_509.mzML?download=1 -O proteomics/data/160920_SM-AKTWT_509.mzML\n",
        "!wget https://zenodo.org/records/17549994/files/160928_SM-6ZEWG_423.mzML?download=1 -O proteomics/data/160928_SM-6ZEWG_423.mzML"
      ],
      "metadata": {
        "id": "CW4E3y6biqBC",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ba42ede8-2c92-4a7b-b178-6aab51783e7d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-11-07 11:05:29--  https://zenodo.org/records/17549994/files/160513-SM-AHYMJ-14.mzML?download=1\n",
            "Resolving zenodo.org (zenodo.org)... 188.185.43.153, 188.185.48.75, 137.138.52.235, ...\n",
            "Connecting to zenodo.org (zenodo.org)|188.185.43.153|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 461672605 (440M) [application/octet-stream]\n",
            "Saving to: â€˜proteomics/data/160513-SM-AHYMJ-14.mzMLâ€™\n",
            "\n",
            "proteomics/data/160 100%[===================>] 440.29M  1.80MB/s    in 2m 17s  \n",
            "\n",
            "2025-11-07 11:07:46 (3.22 MB/s) - â€˜proteomics/data/160513-SM-AHYMJ-14.mzMLâ€™ saved [461672605/461672605]\n",
            "\n",
            "--2025-11-07 11:07:47--  https://zenodo.org/records/17549994/files/160914_SM-9SD9G_284.mzML?download=1\n",
            "Resolving zenodo.org (zenodo.org)... 188.185.48.75, 137.138.52.235, 188.185.43.153, ...\n",
            "Connecting to zenodo.org (zenodo.org)|188.185.48.75|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 311793888 (297M) [application/octet-stream]\n",
            "Saving to: â€˜proteomics/data/160914_SM-9SD9G_284.mzMLâ€™\n",
            "\n",
            "proteomics/data/160 100%[===================>] 297.35M  3.34MB/s    in 5m 35s  \n",
            "\n",
            "2025-11-07 11:13:22 (910 KB/s) - â€˜proteomics/data/160914_SM-9SD9G_284.mzMLâ€™ saved [311793888/311793888]\n",
            "\n",
            "--2025-11-07 11:13:23--  https://zenodo.org/records/17549994/files/160920_SM-AKTWT_509.mzML?download=1\n",
            "Resolving zenodo.org (zenodo.org)... 188.185.48.75, 137.138.52.235, 188.185.43.153, ...\n",
            "Connecting to zenodo.org (zenodo.org)|188.185.48.75|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 253910025 (242M) [application/octet-stream]\n",
            "Saving to: â€˜proteomics/data/160920_SM-AKTWT_509.mzMLâ€™\n",
            "\n",
            "proteomics/data/160 100%[===================>] 242.15M  3.52MB/s    in 2m 42s  \n",
            "\n",
            "2025-11-07 11:16:05 (1.50 MB/s) - â€˜proteomics/data/160920_SM-AKTWT_509.mzMLâ€™ saved [253910025/253910025]\n",
            "\n",
            "--2025-11-07 11:16:05--  https://zenodo.org/records/17549994/files/160928_SM-6ZEWG_423.mzML?download=1\n",
            "Resolving zenodo.org (zenodo.org)... 137.138.52.235, 188.185.48.75, 188.185.43.153, ...\n",
            "Connecting to zenodo.org (zenodo.org)|137.138.52.235|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 340328847 (325M) [application/octet-stream]\n",
            "Saving to: â€˜proteomics/data/160928_SM-6ZEWG_423.mzMLâ€™\n",
            "\n",
            "proteomics/data/160 100%[===================>] 324.56M  1.55MB/s    in 2m 59s  \n",
            "\n",
            "2025-11-07 11:19:05 (1.82 MB/s) - â€˜proteomics/data/160928_SM-6ZEWG_423.mzMLâ€™ saved [340328847/340328847]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! nextflow pull nf-core/quantms"
      ],
      "metadata": {
        "id": "v6ICyhGxilCO",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "dfb85524-f56f-4247-a235-cfaf79e79272"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Checking nf-core/quantms ...\n",
            " downloaded from https://github.com/nf-core/quantms.git - revision: 38e08da5b4 [master]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "CG0i4NLAwZmX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! nextflow run nf-core/quantms --input ./proteomics/data/sample_sheet.tsv --database ./proteomics/databases/proteindatabase.fasta --outdir ./proteomics/results -profile conda -c proteomics/low_resources.config -resume"
      ],
      "metadata": {
        "id": "hXGOe4ctjP-i",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "01385d51-43c7-42a6-fe13-f73437571447"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "curl: /usr/local/lib/libcurl.so.4: no version information available (required by curl)\n",
            "\n",
            "\u001b[1m\u001b[38;5;232m\u001b[48;5;43m N E X T F L O W \u001b[0;2m  ~  \u001b[mversion 25.10.0\u001b[m\n",
            "\u001b[K\n",
            "Launching\u001b[35m `https://github.com/nf-core/quantms` \u001b[0;2m[\u001b[0;1;36mcompassionate_noether\u001b[0;2m] DSL2 - \u001b[36mrevision: \u001b[0;36m38e08da5b4 [master]\u001b[m\n",
            "\u001b[K\n",
            "\u001b[33mWARN: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
            "  There is a problem with your Conda configuration!\n",
            "\n",
            "  You will need to set-up the conda-forge and bioconda channels correctly.\n",
            "  Please refer to https://bioconda.github.io/\n",
            "  The observed channel order is \n",
            "  [defaults, conda-forge, bioconda]\n",
            "  but the following channel order is required:\n",
            "  [conda-forge, bioconda, defaults]\n",
            "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[39m\u001b[K\n",
            "\n",
            "\n",
            "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
            "                                        \u001b[0;32m,--.\u001b[0;30m/\u001b[0;32m,-.\u001b[0m\n",
            "\u001b[0;34m        ___     __   __   __   ___     \u001b[0;32m/,-._.--~'\u001b[0m\n",
            "\u001b[0;34m  |\\ | |__  __ /  ` /  \\ |__) |__         \u001b[0;33m}  {\u001b[0m\n",
            "\u001b[0;34m  | \\| |       \\__, \\__/ |  \\ |___     \u001b[0;32m\\`-._,-`-,\u001b[0m\n",
            "                                        \u001b[0;32m`._,._,'\u001b[0m\n",
            "\u001b[0;35m  nf-core/quantms v1.2.0-g38e08da\u001b[0m\n",
            "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
            "\u001b[1mCore Nextflow options\u001b[0m\n",
            "  \u001b[0;34mrevision                       : \u001b[0;32mmaster\u001b[0m\n",
            "  \u001b[0;34mrunName                        : \u001b[0;32mcompassionate_noether\u001b[0m\n",
            "  \u001b[0;34mlaunchDir                      : \u001b[0;32m/content\u001b[0m\n",
            "  \u001b[0;34mworkDir                        : \u001b[0;32m/content/work\u001b[0m\n",
            "  \u001b[0;34mprojectDir                     : \u001b[0;32m/root/.nextflow/assets/nf-core/quantms\u001b[0m\n",
            "  \u001b[0;34muserName                       : \u001b[0;32mroot\u001b[0m\n",
            "  \u001b[0;34mprofile                        : \u001b[0;32mconda\u001b[0m\n",
            "  \u001b[0;34mconfigFiles                    : \u001b[0;32m\u001b[0m\n",
            "\n",
            "\u001b[1mInput/output options\u001b[0m\n",
            "  \u001b[0;34minput                          : \u001b[0;32m./proteomics/data/sample_sheet.tsv\u001b[0m\n",
            "  \u001b[0;34moutdir                         : \u001b[0;32m./proteomics/results\u001b[0m\n",
            "  \u001b[0;34macquisition_method             : \u001b[0;32mnull\u001b[0m\n",
            "\n",
            "\u001b[1mProtein database\u001b[0m\n",
            "  \u001b[0;34mdatabase                       : \u001b[0;32m./proteomics/databases/proteindatabase.fasta\u001b[0m\n",
            "\n",
            "\u001b[1mDatabase search\u001b[0m\n",
            "  \u001b[0;34minstrument                     : \u001b[0;32mnull\u001b[0m\n",
            "\n",
            "\u001b[1mModification localization\u001b[0m\n",
            "  \u001b[0;34mluciphor_debug                 : \u001b[0;32m0\u001b[0m\n",
            "\n",
            "\u001b[1mPSM re-scoring (general)\u001b[0m\n",
            "  \u001b[0;34mrun_fdr_cutoff                 : \u001b[0;32m0.10\u001b[0m\n",
            "\n",
            "\u001b[1mPSM re-scoring (Percolator)\u001b[0m\n",
            "  \u001b[0;34mdescription_correct_features   : \u001b[0;32m0\u001b[0m\n",
            "\n",
            "\u001b[1mConsensus ID\u001b[0m\n",
            "  \u001b[0;34mconsensusid_considered_top_hits: \u001b[0;32m0\u001b[0m\n",
            "  \u001b[0;34mmin_consensus_support          : \u001b[0;32m0\u001b[0m\n",
            "\n",
            "\u001b[1mIsobaric analyzer\u001b[0m\n",
            "  \u001b[0;34mselect_activation              : \u001b[0;32mHCD\u001b[0m\n",
            "\n",
            "\u001b[1mProtein Quantification (LFQ)\u001b[0m\n",
            "  \u001b[0;34mfeature_with_id_min_score      : \u001b[0;32m0.10\u001b[0m\n",
            "  \u001b[0;34mfeature_without_id_min_score   : \u001b[0;32m0.75\u001b[0m\n",
            "  \u001b[0;34mlfq_intensity_threshold        : \u001b[0;32m1000\u001b[0m\n",
            "\n",
            "\u001b[1mDIA-NN\u001b[0m\n",
            "  \u001b[0;34mscan_window                    : \u001b[0;32m8\u001b[0m\n",
            "\n",
            "\u001b[1mStatistical post-processing\u001b[0m\n",
            "  \u001b[0;34mcontrasts                      : \u001b[0;32mpairwise\u001b[0m\n",
            "\n",
            "\u001b[1mQuality control\u001b[0m\n",
            "  \u001b[0;34menable_pmultiqc                : \u001b[0;32mtrue\u001b[0m\n",
            "  \u001b[0;34mpmultiqc_idxml_skip            : \u001b[0;32mtrue\u001b[0m\n",
            "\n",
            "\u001b[1mMax job request options\u001b[0m\n",
            "  \u001b[0;34mmax_cpus                       : \u001b[0;32m2\u001b[0m\n",
            "  \u001b[0;34mmax_memory                     : \u001b[0;32m6 GB\u001b[0m\n",
            "  \u001b[0;34mmax_time                       : \u001b[0;32m2h\u001b[0m\n",
            "\n",
            "!! Only displaying parameters that differ from the pipeline defaults !!\n",
            "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
            "If you use nf-core/quantms for your analysis please cite:\n",
            "\n",
            "* The nf-core framework\n",
            "  https://doi.org/10.1038/s41587-020-0439-x\n",
            "\n",
            "* Software dependencies\n",
            "  https://github.com/nf-core/quantms/blob/master/CITATIONS.md\n",
            "\n",
            "ERROR ~ Cannot invoke method optional() on null object\n",
            "\n",
            " -- Check script '/root/.nextflow/assets/nf-core/quantms/modules/local/msstatstmt/main.nf' at line: 18 or see '.nextflow.log' file for more details\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "WoL72Fyjr2GC"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}