{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Running Nextflow from Colab\n",
        "\n",
        "This is a guide with code to be able to run nf-core pipelines from colab notebooks."
      ],
      "metadata": {
        "id": "J_AfYl8IOo-q"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "9g5_k5j2OnCg"
      },
      "outputs": [],
      "source": [
        "!apt update\n",
        "!apt install openjdk-17-jdk\n",
        "!export JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64\n",
        "!export PATH=$JAVA_HOME/bin:$PATH\n",
        "!source ~/.bashrc"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Installing Nextflow"
      ],
      "metadata": {
        "id": "ovA2dn4BPIaY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget -qO- https://get.nextflow.io | bash # Download Nextflow\n",
        "!mv nextflow /usr/bin/nextflow # Move to a path Colab can access\n",
        "!chmod +x /usr/bin/nextflow # Make it executable\n",
        "!nextflow -v # Test it"
      ],
      "metadata": {
        "id": "y3Zl5FQNPMPB"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Setting up Conda"
      ],
      "metadata": {
        "id": "t3wv9ZG-PRbU"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -q condacolab # -q here means quite\n",
        "import condacolab\n",
        "condacolab.install()"
      ],
      "metadata": {
        "id": "yYeM8d99PT-z"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!conda config --add channels bioconda\n",
        "!conda config --add channels conda-forge\n",
        "!conda config --add channels defaults\n",
        "!conda config --set channel_priority strict"
      ],
      "metadata": {
        "id": "kk9FMckIPUjp"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Running a Test pipeline"
      ],
      "metadata": {
        "id": "sursu7ahPZSw"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#! nextflow pull nf-core/demo"
      ],
      "metadata": {
        "id": "lMymtYBoO7Eb"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#! nextflow run nf-core/demo -profile conda,test --outdir demo-results"
      ],
      "metadata": {
        "id": "IygCHi6ePtQv"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Running RNAseq pipeline\n",
        "\n",
        "![RNASeq pipeline](https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/transcriptomics/notebooks/img/rnaseq.png)\n",
        "\n",
        "Data downloaded from [GSE137344](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE137344)."
      ],
      "metadata": {
        "id": "z_2UONcTik6t"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Creating Data Folders for Transcriptomics Analysis"
      ],
      "metadata": {
        "id": "oK3h5HyYWa2G"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir transcriptomics\n",
        "!mkdir transcriptomics/data"
      ],
      "metadata": {
        "id": "CZRNWpZ_T2Ke"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! rm -r transcriptomics/results"
      ],
      "metadata": {
        "id": "k08jUxjtX_Mx"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Downloading Sample Sheet File for nf-core/rnaseq"
      ],
      "metadata": {
        "id": "tabULdAGWhtD"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! wget https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/transcriptomics/data/sample_sheet.csv -O transcriptomics/data/sample_sheet.csv"
      ],
      "metadata": {
        "id": "Gyh474X9UtE2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Getting the Config file for the pipeline"
      ],
      "metadata": {
        "id": "Up8J2BeaBkjp"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! wget https://raw.githubusercontent.com/Multiomics-Analytics-Group/course_multi-omics_data_science/refs/heads/main/transcriptomics/low_resources.config -O transcriptomics/low_resources.config"
      ],
      "metadata": {
        "id": "OXl5sLPeVVef"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://trace.ncbi.nlm.nih.gov/Traces/sra-reads-be/fastq?acc=SRR10104255 -O transcriptomics/data/SRR10104255.fastq.gz\n",
        "!wget https://trace.ncbi.nlm.nih.gov/Traces/sra-reads-be/fastq?acc=SRR10104256 -O transcriptomics/data/SRR10104256.fastq.gz\n",
        "!wget https://trace.ncbi.nlm.nih.gov/Traces/sra-reads-be/fastq?acc=SRR10104257 -O transcriptomics/data/SRR10104257.fastq.gz\n",
        "!wget https://trace.ncbi.nlm.nih.gov/Traces/sra-reads-be/fastq?acc=SRR10104258 -O transcriptomics/data/SRR10104258.fastq.gz"
      ],
      "metadata": {
        "id": "CW4E3y6biqBC"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! nextflow pull nf-core/rnaseq"
      ],
      "metadata": {
        "id": "v6ICyhGxilCO"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "CG0i4NLAwZmX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! nextflow run \\\n",
        "    nf-core/rnaseq \\\n",
        "    --input transcriptomics/data/sample_sheet.csv \\\n",
        "    --outdir transcriptomics/results \\\n",
        "    --igenomes_ignore \\\n",
        "    --genome null \\\n",
        "    -profile conda \\\n",
        "    -c transcriptomics/low_resources.config \\\n",
        "    -resume"
      ],
      "metadata": {
        "id": "hXGOe4ctjP-i"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "WoL72Fyjr2GC"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}